sympy/core/containers.py

```diff
@@ -144,6 +145,19 @@ def index(self, value, start=None, stop=None):
         else:
             return self.args.index(value, start, stop)
 
+    @property
+    def kind(self):
+        """
+        kind method is used to determine type of Tuple objects
+
+        Examples
+        ========
+
+        >>> from sympy import Tuple, NumberKind
+        >>> Tuple(1,2).kind
+        TupleKind(NumberKind, NumberKind)
+        """
+        return TupleKind(tuple(i.kind for i in self.args))
```

Comment:
These should be separate arguments e.g. `TupleKind(*(i.kind for i in self.args))`

Reply:
it throws TypeError `TypeError: not all arguments converted during string formatting`,
current code kind of make sense to me
```python
class TupleKind(Kind):
    """
    TupleKind is subclass of Kind which is used to find kind of Tuple

    Examples
    ========

    >>> from sympy import Tuple
    >>> Tuple(1,2).kind
    TupleKind(NumberKind, NumberKind)

    """
    def __new__(cls, *args):
        obj = super().__new__(cls, *args)
        obj.element_kind = args
        return obj

    def __repr__(self):
        return "TupleKind%s" % self.element_kind
```

Reply:
The `__repr__` method also needs to be changed.